cmd/snap/cmd_snap_op.go

```diff
@@ -353,6 +381,60 @@ func (x *cmdInstall) Execute([]string) error {
 	return showDone([]string{name}, "install")
 }
 
+func (x *cmdInstall) installMany(names []string) error {
+	// sanity check
+	for _, name := range names {
+		if strings.Contains(name, "/") || strings.HasSuffix(name, ".snap") || strings.Contains(name, ".snap.") {
+			return fmt.Errorf("only a single snap file can be installed")
+		}
+	}
+
+	cli := Client()
+	changeID, err := cli.InstallMany(names)
+	if err != nil {
+		return err
+	}
+
+	chg, err := wait(cli, changeID)
+	if err != nil {
+		return err
+	}
+
+	var installed []string
+	if err := chg.Get("snap-names", &installed); err != nil && err != client.ErrNoData {
+		return err
+	}
+
+	if len(installed) > 0 {
+		return showDone(installed, "install")
+	}
+
+	return nil
+}
+
+func (x *cmdInstall) Execute([]string) error {
+	if err := x.setChannelFromCommandline(); err != nil {
+		return err
+	}
+	if err := x.validateMode(); err != nil {
+		return err
+	}
+
+	dangerous := x.Dangerous || x.ForceDangerous
+	if len(x.Positional.Snaps) == 1 {
+		opts := &client.SnapOptions{
+			Channel:   x.Channel,
+			DevMode:   x.DevMode,
+			JailMode:  x.JailMode,
+			Revision:  x.Revision,
+			Dangerous: dangerous,
```

Comment:
I agree channel might be supported, but it's not clear to me what you say is not done. The point I've raised above is that the options are not being shipped to the server on multi-snap operations for validation, which means they'd feel like accepted but be completely ignored. The point I've made above was to ship these options to the server, just as they'd anyway, so that we get the benefit of validation that is already in place in snapsOp.

Am I missing something?